src/main/app/src/reducers/hakutulosSlice.js

```diff
@@ -0,0 +1,330 @@
+          _.size(koulutusData.hits) === 0 &&
+          _.size(oppilaitosData.hits) > 0 &&
+          isNewKeyword
+        ) {
+          state.selectedTab = OPPILAITOS;
+        } else if (_.size(koulutusData.hits) > 0 && isNewKeyword) {
+          state.selectedTab = KOULUTUS;
+        }
+        if (isReload) {
+          _.forEach(literals, (val, key) => {
+            state[key] = val;
+          });
+          _.forEach(filters, (idsStr, key) => {
+            if (key === 'sijainti') {
+              const maakunnatIds = _.split(idsStr, ',').filter((id) =>
+                _.startsWith(id, 'maakunta_')
+              );
+              const kunnatIds = _.split(idsStr, ',').filter((id) =>
+                _.startsWith(id, 'kunta_')
+              );
+              if (_.size(maakunnatIds) > 0) {
+                state.sijainti = getCheckedFilterValues(
+                  maakunnatIds.join(','),
+                  koulutusData?.filters?.maakunta
+                );
+              }
+              if (_.size(kunnatIds) > 0) {
+                state.selectedSijainti = getSelectedKunnatFilterValues(
+                  kunnatIds,
+                  koulutusData?.filters?.kunta
+                );
+              }
+            } else if (
+              key === 'koulutustyyppi' &&
+              _.startsWith(idsStr, 'koulutustyyppi_')
+            ) {
+              state[key] = getCheckedFilterValues(
+                idsStr,
+                koulutusData.filters?.[key]?.amm?.alakoodit
+              );
+            } else {
+              state[key] = getCheckedFilterValues(idsStr, koulutusData.filters[key]);
+            }
+          });
+        }
+
+        state.loading -= 1;
+        state.error = null;
+        state.status = IDLE_STATUS;
+      }
+    },
+    searchKoulutuksetSuccess(state, { payload }) {
+      if (state.status === LOADING_STATUS) {
+        const { koulutusData } = payload;
+        state.koulutusHits = koulutusData.hits;
+        state.koulutusCount = _.size(koulutusData.hits);
+        state.loading -= 1;
+        state.error = null;
+        state.status = IDLE_STATUS;
+      }
+    },
+    searchOppilaitoksetSuccess(state, { payload }) {
+      if (state.status === LOADING_STATUS) {
+        const { oppilaitosData } = payload;
+        state.oppilaitosHits = oppilaitosData.hits;
+        state.oppilaitosCount = _.size(oppilaitosData.hits);
+        state.loading -= 1;
+        state.error = null;
+        state.status = IDLE_STATUS;
+      }
+    },
+    searchAPICallError: fetchFail,
+  },
+});
+
+export const {
+  setKeyword,
+  setKeywordEditMode,
+  setSelectedTab,
+  searchAPICallStart,
+  searchAPICallError,
+  setOpetuskieli,
+  setKoulutustyyppi,
+  setKoulutusala,
+  setSijainti,
+  setSelectedSijainti,
+  clearOffsetAndPaging,
+  clearSelectedFilters,
+  setSelectedFilters,
+  setOrder,
+  setSort,
+  toggleshowHakutulosFilters,
+  setSize,
+  setKoulutusOffsetAndPage,
+  setOppilaitosOffsetAndPage,
```

Comment:
"And"-funktion nimessä on "code smell". https://medium.com/@samanthaming/code-smell-no-and-in-function-name-c83d0fe5d46e